src/relax/ir/expr_functor.cc

```diff
@@ -752,6 +752,7 @@ Var ExprMutator::VisitVarDef(const Var& var) {
   } else {
     LOG(FATAL) << "TypeError: Invalid type: " << var->GetTypeKey();
   }
+  this->var_remap_[var->vid] = ret;
```

Comment:
I am not too sure if we wantr to do directly remap change here, ideally such remap should be changed in the caller side, either function def or binding

Reply:
We only call `VisitVarDef` at the definition side, e.g. the param or binding. If we mutate the var `x` to `x'` at the definition side, all use side needs to be updated, since `x` no longer exists. 

Could you please show an example to indicate what I missed?

Reply:
After offline discussion, we agree we move the logic to the `VisitExpr_(FunctionNode*)`, thanks again for pointing it out